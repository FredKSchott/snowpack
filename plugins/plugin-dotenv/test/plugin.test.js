const path = require('path');
const execa = require('execa');

const execPluginFilePath = require.resolve('./execPlugin');
const testWorkDirectory = path.join(__dirname, 'env');

function execPlugin(nodeEnv, params = null) {
  const {stdout} = execa.sync('node', [execPluginFilePath, JSON.stringify(params)], {
    cwd: testWorkDirectory,
    env: {NODE_ENV: nodeEnv},
  });
  return JSON.parse(stdout);
}

beforeEach(() => {
  // test that pre-exiting environment variables are not overwritten
  process.env.__DOTENV_PRESET = 'PRESET';
});

const NODE_ENV_LIST = [undefined, 'development', 'test', 'production'];
describe('NODE_ENV=', () => {
  NODE_ENV_LIST.forEach((nodeEnv) => {
    test(`${nodeEnv}`, () => {
      expect(execPlugin(nodeEnv)).toMatchSnapshot();
    });
  });
});

describe('with dir NODE_ENV=', () => {
  NODE_ENV_LIST.forEach((nodeEnv) => {
    test(`${nodeEnv}`, () => {
      expect(execPlugin(nodeEnv, {dir: 'subdir'})).toMatchSnapshot();
    });
  });
});
