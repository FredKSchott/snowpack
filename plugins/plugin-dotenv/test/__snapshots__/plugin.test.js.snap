// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NODE_ENV= development 1`] = `
Object {
  "__DOTENV_DEVELOPMENT": "DEVELOPMENT",
  "__DOTENV_DEVELOPMENT_LOCAL": "DEVELOPMENT_LOCAL",
  "__DOTENV_ENV": "ENV",
  "__DOTENV_LOCAL": "LOCAL",
  "__DOTENV_PRESET": "PRESET",
  "__DOTENV_PRODUCTION": "ENV",
  "__DOTENV_PRODUCTION_LOCAL": "LOCAL",
  "__DOTENV_TEST": "ENV",
  "__DOTENV_TEST_LOCAL": "LOCAL",
}
`;

exports[`NODE_ENV= production 1`] = `
Object {
  "__DOTENV_DEVELOPMENT": "ENV",
  "__DOTENV_DEVELOPMENT_LOCAL": "LOCAL",
  "__DOTENV_ENV": "ENV",
  "__DOTENV_LOCAL": "LOCAL",
  "__DOTENV_PRESET": "PRESET",
  "__DOTENV_PRODUCTION": "PRODUCTION",
  "__DOTENV_PRODUCTION_LOCAL": "PRODUCTION_LOCAL",
  "__DOTENV_TEST": "ENV",
  "__DOTENV_TEST_LOCAL": "LOCAL",
}
`;

exports[`NODE_ENV= test 1`] = `
Object {
  "__DOTENV_DEVELOPMENT": "ENV",
  "__DOTENV_DEVELOPMENT_LOCAL": "ENV",
  "__DOTENV_ENV": "ENV",
  "__DOTENV_LOCAL": "TEST",
  "__DOTENV_PRESET": "PRESET",
  "__DOTENV_PRODUCTION": "ENV",
  "__DOTENV_PRODUCTION_LOCAL": "ENV",
  "__DOTENV_TEST": "TEST",
  "__DOTENV_TEST_LOCAL": "TEST_LOCAL",
}
`;

exports[`NODE_ENV= undefined 1`] = `
Object {
  "__DOTENV_DEVELOPMENT": "ENV",
  "__DOTENV_DEVELOPMENT_LOCAL": "LOCAL",
  "__DOTENV_ENV": "ENV",
  "__DOTENV_LOCAL": "LOCAL",
  "__DOTENV_PRESET": "PRESET",
  "__DOTENV_PRODUCTION": "ENV",
  "__DOTENV_PRODUCTION_LOCAL": "LOCAL",
  "__DOTENV_TEST": "ENV",
  "__DOTENV_TEST_LOCAL": "LOCAL",
}
`;

exports[`with dir NODE_ENV= development 1`] = `
Object {
  "__DOTENV_DEVELOPMENT": "DEVELOPMENT",
  "__DOTENV_DEVELOPMENT_LOCAL": "DEVELOPMENT_LOCAL",
  "__DOTENV_ENV": "ENV",
  "__DOTENV_LOCAL": "LOCAL",
  "__DOTENV_PRESET": "PRESET",
  "__DOTENV_PRODUCTION": "ENV",
  "__DOTENV_PRODUCTION_LOCAL": "LOCAL",
  "__DOTENV_TEST": "ENV",
  "__DOTENV_TEST_LOCAL": "LOCAL",
}
`;

exports[`with dir NODE_ENV= production 1`] = `
Object {
  "__DOTENV_DEVELOPMENT": "ENV",
  "__DOTENV_DEVELOPMENT_LOCAL": "LOCAL",
  "__DOTENV_ENV": "ENV",
  "__DOTENV_LOCAL": "LOCAL",
  "__DOTENV_PRESET": "PRESET",
  "__DOTENV_PRODUCTION": "PRODUCTION",
  "__DOTENV_PRODUCTION_LOCAL": "PRODUCTION_LOCAL",
  "__DOTENV_TEST": "ENV",
  "__DOTENV_TEST_LOCAL": "LOCAL",
}
`;

exports[`with dir NODE_ENV= test 1`] = `
Object {
  "__DOTENV_DEVELOPMENT": "ENV",
  "__DOTENV_DEVELOPMENT_LOCAL": "ENV",
  "__DOTENV_ENV": "ENV",
  "__DOTENV_LOCAL": "TEST",
  "__DOTENV_PRESET": "PRESET",
  "__DOTENV_PRODUCTION": "ENV",
  "__DOTENV_PRODUCTION_LOCAL": "ENV",
  "__DOTENV_TEST": "TEST",
  "__DOTENV_TEST_LOCAL": "TEST_LOCAL",
}
`;

exports[`with dir NODE_ENV= undefined 1`] = `
Object {
  "__DOTENV_DEVELOPMENT": "ENV",
  "__DOTENV_DEVELOPMENT_LOCAL": "LOCAL",
  "__DOTENV_ENV": "ENV",
  "__DOTENV_LOCAL": "LOCAL",
  "__DOTENV_PRESET": "PRESET",
  "__DOTENV_PRODUCTION": "ENV",
  "__DOTENV_PRODUCTION_LOCAL": "LOCAL",
  "__DOTENV_TEST": "ENV",
  "__DOTENV_TEST_LOCAL": "LOCAL",
}
`;
